server/image_pull.go

```diff
@@ -298,13 +310,9 @@ func (s *Server) pullImage(ctx context.Context, pullArgs *pullArguments) (string
 	// Update metric for successful image pulls
 	metrics.Instance().MetricImagePullsSuccessesInc(pulled)
 
-	status, err := s.StorageImageServer().ImageStatus(s.config.SystemContext, pulled)
+	imageRef, err := s.StorageImageServer().PullLocalManifestDigest(ctx, &sourceCtx, pulled)
```

Comment:
`PullImage` already has that data internally. This discards that data, and effectively does a whole new pull, with no progress reporting etc.

Among other things, that makes the “image is already in store” optimization completely pointless, because that one tries to avoid issuing even a metadata-only pull.

Reply:
Alright, I moved the logic into `PullImage` of the container server. This way we can directly retrieve the `manifestBytes` there.